scripts/detect_artefacts.py

```diff
@@ -0,0 +1,311 @@
+    for ind, val in enumerate(test_list):
+        im_read = cv2.imread(os.path.join(root_dir, val))
+        kold = im_read.copy()
+        kold = cv2.cvtColor(kold, cv2.COLOR_BGR2GRAY)
+        imm = cv2.cvtColor(im_read, cv2.COLOR_BGR2RGB)
+        imm = imm / 255
+        imm2 = [torch.tensor(imm, dtype=torch.float32).permute(2, 0, 1)]
+        pred = artefact(imm2)
+        tg, tg2, tg1 = post_nms(pred, imm2)  # box, score, label
+        tg = [list(i) for i in tg]
+        bbox.append(tg)
+        art_class.append(tg1)
+        c_score.append(tg2)
+        if save_image:
+            for ind_2, val_2 in enumerate(tg):
+                ji = {'1': [(0, 150, 150)], '2': [(0, 0, 0)], '3': [(0, 150, 0)], '4': [(150, 0, 0)]}
+                juj = {'1': "QR_Code", "2": "Bar_Code", "3": "Logo", "4": "Photo"}
+                cv2.rectangle(imm, (int(val_2[0]), int(val_2[1])), (int(val_2[2]), int(val_2[3])),
+                              ji[str(int(tg1[ind_2]))][0], 2)
+                imm = cv2.putText(imm, str(tg2[ind_2]), (int(val_2[0]), int(val_2[3]) + 5), cv2.FONT_HERSHEY_SIMPLEX,
+                                  1.5,
+                                  ji[str(int(tg1[ind_2]))][0], 2, )
+                imm = cv2.putText(imm, juj[str(int(tg1[ind_2]))], (int(val_2[0]), int(val_2[1])),
+                                  cv2.FONT_HERSHEY_SIMPLEX,
+                                  1.5,
+                                  ji[str(int(tg1[ind_2]))][0], 2, )
+            theimagee = Image.fromarray((imm * 255).astype(np.uint8))
+            theimagee.save(f'{save_to_folder}/{val}')
+        if code_KIE:
+            for i in range(len(tg)):
+                txt = []
+                if int(tg1[i]) == 1:
+                    pio = kold[int(tg[i][1]) - 6:int(tg[i][3]) + 6, int(tg[i][0]) - 6:int(tg[i][2]) + 6]
+                    try:
+                        txt = reader.decode_array(pio)
+                        if "parsed" in list(txt[0].keys()):
+                            code_info.append(txt[0]["parsed"].decode())
+                        else:
+                            code_info.append(["Unreadable/Not a code"])
+                    except ZeroDivisionError:
+                        code_info.append(["Unreadable/Not a code"])
+
+                elif int(tg1[i]) == 2:
+                    pio = kold[int(tg[i][1]) + 10:int(tg[i][3]) - 5, int(tg[i][0]) - 2:int(tg[i][2]) + 2]
+                    try:
+                        txt = decode(pio)
+                        code_info.append(txt)
+                    except KeyError:
+                        code_info.append(["Unreadable/Not a code"])
+
+            inf_dic.update(
+                {"geometry": str(bbox), "class": str(art_class), "score": str(c_score), "KIE": str(code_info)})
+        else:
+            inf_dic.update({"geometry": str(bbox), "class": str(art_class), "score": str(c_score)})
+    return inf_dic
+
+
+def main(args):
+    torch.backends.cudnn.benchmark = True
```

Comment:
This is an inference script, we won't forward enough time to benchmark cudnn ;)

Reply:
Yes, alright